src/backend/gl/src/factory.rs

```diff
@@ -436,12 +436,13 @@ impl f::Factory<R> for Factory {
         Ok(self.share.handles.borrow_mut().make_texture(object, desc))
     }
 
-    fn view_buffer_as_shader_resource_raw(&mut self, hbuf: &handle::RawBuffer<R>)
+    fn view_buffer_as_shader_resource_raw(&mut self, hbuf: &handle::RawBuffer<R>, format: Format)
                                       -> Result<handle::RawShaderResourceView<R>, f::ResourceViewError> {
         let gl = &self.share.context;
         let mut name = 0 as gl::types::GLuint;
         let buf_name = *self.frame_handles.ref_buffer(hbuf);
-        let format = gl::R8; //TODO: get from the buffer handle
+        let format = tex::format_to_glfull(format)
+            .map_err(|_| f::ResourceViewError::Unsupported)?;
```

Comment:
too much sugar for my taste, given that a single `match` would be almost as concise